src/batch-client/index.ts

```diff
@@ -1185,8 +1185,10 @@ export class ZimbraBatchClient {
 
 			this.checkAndUpdateSessionId(sessionId);
 
-			if (notifications && this.notificationHandler) {
-				this.notificationHandler(notifications);
+			if (notifications) {
+				setTimeout(() => {
```

Comment:
can you please add an inline comment describing the reason ?

Reply:
added